src/com/sap/piper/versioning/MavenArtifactVersioning.groovy

```diff
@@ -0,0 +1,19 @@
+package com.sap.piper.versioning
+
+class MavenArtifactVersioning extends ArtifactVersioning {
+    protected MavenArtifactVersioning (script, configuration) {
+        super(script, configuration)
+    }
+
+    @Override
+    def getVersion() {
+        def mavenPom = script.readMavenPom (file: configuration.filePath)
+        return mavenPom.getVersion().replaceAll(/-SNAPSHOT$/, "")
+    }
+
+    @Override
+    def setVersion(version) {
+        def mvnParameter = configuration.filePath == 'pom.xml' ? '' : "--file ${configuration.filePath}"
+        script.sh "mvn versions:set -DnewVersion=${version} ${mvnParameter}"
```

Comment:
here we could also omit the condition and provide `--file` for all cases

Reply:
done